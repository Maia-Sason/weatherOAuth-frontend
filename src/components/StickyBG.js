import { Box, makeStyles, Fade, Grow } from "@material-ui/core";
import { PaperBG } from "./PaperBG";

const StickyBG = ({ icon, load }) => {
  const image = PaperBG(icon);
  const useStyles = makeStyles(() => ({
    background: {
      backgroundImage: `url(${image})`,
      height: "100vh",
      width: "100vw",
      backgroundSize: "cover",
      backgroundColor: "black",
      position: "fixed",
      top: 0,
      height: "100vh",
    },
    container: {
      backgroundColor: "black",
      height: "100%",
      width: "100%",
      backgroundSize: "cover",
      position: "fixed",
      top: 0,
      zIndex: -2,
    },
  }));

  const classes = useStyles();

  return (
    <>
      <Box className={classes.container}>
        fdassdf
        <Fade in={load} {...(load ? { setTimeout: 10000 } : {})}>
          <Box className={classes.background}></Box>
        </Fade>
      </Box>
    </>
  );
};

export default StickyBG;
