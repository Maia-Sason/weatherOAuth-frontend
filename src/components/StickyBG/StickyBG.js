import { Box, makeStyles, Fade, Grow } from "@material-ui/core";
import { PaperBG } from "../PaperBG";

const StickyBG = ({ icon, load }) => {
  const image = PaperBG(icon);
  const useStyles = makeStyles(() => ({
    background: {
      backgroundImage: `url(${image})`,
      height: "100vh",
      width: "100vw",
      backgroundSize: "cover",
      backgroundColor: "black",
      position: "fixed",
      top: 0,
    },
    container: {
      backgroundColor: "black",
      height: "100%",
      width: "100%",
      backgroundSize: "cover",
      position: "fixed",
      top: 0,
      zIndex: -2,
    },
  }));

  const classes = useStyles();

  return (
    <>
      <Box className={classes.container}>
        <Fade in={load} {...(load ? { settimeout: 10000 } : {})}>
          <Box className={classes.background}></Box>
        </Fade>
      </Box>
    </>
  );
};

export default StickyBG;
