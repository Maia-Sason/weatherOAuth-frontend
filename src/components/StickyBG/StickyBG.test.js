import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import StickyBG from "./StickyBG";

test("<StickyBG /> matches snapshot", () => {
  const component = render(<StickyBG icon={"10d"} load={true} />);
  expect(component.container).toMatchSnapshot();
});

it("renders StickyBG without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<StickyBG icon={"10d"} load={true} />, div);
});
