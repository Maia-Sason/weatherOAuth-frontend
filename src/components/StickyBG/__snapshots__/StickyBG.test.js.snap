// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StickyBG /> matches snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3 makeStyles-container-2"
  >
    <div
      class="MuiBox-root MuiBox-root-4 makeStyles-background-1"
      settimeout="10000"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
  </div>
</div>
`;
